### PR TITLE
adding logging of training loss

### PR DESCRIPTION
**Pull request by liususan091219**
adding logging of training loss
adding logging of validation loss

### CODE REVIEW

File: flaml/nlp/utils.py [hunk 1]
```diff
@@ -272,7 +274,7 @@ def tokenize_text_multiplechoice(X, custom_hpo_args):
     return output, None
 
 
-def tokenize_swag(this_row, this_tokenizer, custom_hpo_args):
+def tokenize_swag(this_row, this_tokenizer, custom_hpo_args=None):
```
Comment (sonichi) [inline flaml/nlp/utils.py:277]: rename all "this_tokenizer"
```suggestion
def tokenize_swag(this_row, tokenizer, custom_hpo_args=None):
```

File: flaml/nlp/huggingface/trainer.py [hunk 2]
```diff
@@ -44,6 +44,21 @@ def prediction_step(
...
+            super().log(logs)
+        else:
+            super(Seq2SeqTrainer, self).log(logs)
+        if not hasattr(self, "intermediate_results"):
+            self.intermediate_results = {}
+        try:
+            epoch_num = logs["epoch"]
+            for each_key, each_val in logs.items():
+                self.intermediate_results.setdefault(epoch_num, {})
+                self.intermediate_results[epoch_num][each_key] = each_val
```
Comment (sonichi) [inline flaml/nlp/huggingface/trainer.py:58]:
```suggestion
            self.intermediate_results.setdefault(epoch_num, {})
            self.intermediate_results[epoch_num].update(logs)
```

File: flaml/nlp/huggingface/trainer.py [hunk 3]
```diff
@@ -44,6 +44,21 @@ def prediction_step(
...
+        except KeyError:
+            pass
+
```
Comment (sonichi) [inline flaml/nlp/huggingface/trainer.py:61]: remove

**Review by sonichi — APPROVED** [review]
